### PR TITLE
Add azure function discovery

### PR DESCRIPTION
**Pull request by garuppel**

### CODE REVIEW

File: service/discovery/azure/compute.go [hunk 1]
```diff
@@ -74,9 +75,52 @@ func (d *azureComputeDiscovery) List() (list []voc.IsResource, err error) {
...
+	list = append(list, function...)
+
 	return
 }
 
+// Discover function
+func (d *azureComputeDiscovery) discoverFunction() ([]voc.IsResource, error) {
+	var list []voc.IsResource
+
+	client := web.NewAppsClient(to.String(d.sub.SubscriptionID))
```
Comment (oxisto) [inline service/discovery/azure/compute.go:92]: Are you sure, this is the correct call for Azure functions? Isn't this referring to https://azure.microsoft.com/en-us/services/app-service/web/?

Comment (garuppel) [reply service/discovery/azure/compute.go:92]: honestly I do not know :) The call above gets the Function App shown here: https://portal.azure.com/#blade/HubsExtension/BrowseResource/resourceType/Microsoft.Web%2Fsites/kind/functionapp

I don´t know if and what is the difference between Azure Functions and Function Apps.

Comment (garuppel) [reply service/discovery/azure/compute.go:92]: You´re right, functions are created in function apps. Ok, than I have to extend that.

Comment (oxisto) [reply service/discovery/azure/compute.go:92]: It looks like it is the correct call, but it seems they are grouped in "App Service Plans", they seem to be be a container for those apps, but I think we can ignore that for now, so it seems that everything looks good for now.

**Review by oxisto — APPROVED** [review]
LGTM

**Comment by oxisto** [comment]
Ah... you branched it off from the `resource-graph` branch instead of `main`, @garuppel, mistake?

**Comment by garuppel** [comment]
> Ah... you branched it off from the `resource-graph` branch instead of `main`, @garuppel, mistake? Can you re-open a PR/branch from main?
No, I thought that the functions also have references and wanted to take the branch with the implemented references stuff.

**Pull request merged**